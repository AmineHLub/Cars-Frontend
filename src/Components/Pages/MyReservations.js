import { useEffect } from 'react';
// import { useDispatch, useSelector } from 'react-redux';
// import { createReservationAction } from '../../Redux/Reservation';

const MyReservations = () => {
  // const reservationState = useSelector((state) => state.reservationReducer.reservations);
  // const userState = useSelector((state) => state.userReducer);
  // const carState = useSelector((state) => state.carReducer);
  // const dispatch = useDispatch();

  // const onSubmit = (data) => {
  //   const reserveData = {
  //     car_id: selectedCar,
  //     user_id: userState.id,
  //     start_date: selectedDate,
  //     duration: data.duration,
  //   };

  //   // dispatch the data to the create action
  //   dispatch(createReservationAction(reserveData));
  // };

  useEffect(() => {}, []);

  return (
    <div className="absolute h-screen bg-primaryGreen w-full">
      <div className="bg-show-car opacity-20" />
      <div className="text-white opacity" />
      <div className="absolute top-0 h-screen w-full flex items-start justify-center">
        <div className="py-4 px-4 flex flex-col items-center justify-center">
          <div className="text-center p-4 w-full md:w-2/3">
            <h3 className="text-white text-lg md:text-2xl md:mb-10 font-bold">
              YOUR RESERVED CARS
            </h3>
            <p className="text-white mt-5 md:text-xl">
              You can manage your cars here.
            </p>
          </div>
          <div className="p-1 border overflow-hidden border-white rounded-md w-auto mx-4 md:w-1/2 md:mt-10">
            <div className="overflow-x-auto">
              <div className="shadow-md bg-white w-auto flex justify-center mx-auto">
                <table className="w-auto flex flex-col">
                  <thead>
                    <tr className="flex space-x-5 w-full h-auto bg-gray-800 p-2 text-gray-200">
                      <th className="text-sm">Name</th>
                      <th className="text-sm">Description</th>
                      <th className="text-sm">Start Date</th>
                      <th className="text-sm">End Date</th>
                      <th className="text-sm">Duration</th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr className="flex space-x-5 w-full p-2">
                      <td>Name</td>
                      <td>Name</td>
                      <td>Name</td>
                      <td>Name</td>
                      <td>duration</td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>

          </div>
        </div>
      </div>
    </div>
  );
};

export default MyReservations;
