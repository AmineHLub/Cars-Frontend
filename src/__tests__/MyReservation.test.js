import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import MyReservations from '../components/Pages/MyReservations';

describe('Testing My Reservation component', () => {
  test('renders a correct data', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router><MyReservations /></Router>
      </Provider>,
    );
    expect(getByTestId('h3')).toBeInTheDocument();
    expect(getByTestId('p')).toBeInTheDocument();
    expect(getByTestId('table')).toBeInTheDocument();
  });

  test('renders as expected', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router><MyReservations /></Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
