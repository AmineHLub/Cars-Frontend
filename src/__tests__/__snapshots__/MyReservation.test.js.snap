// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing My Reservation component renders as expected 1`] = `
<div
  className="absolute h-screen bg-primaryGreen w-full"
>
  <div
    className="bg-show-car opacity-20"
  />
  <div
    className="text-white opacity"
  />
  <div
    className="absolute top-0 h-screen w-full"
  >
    <div
      className="py-4 px-4 flex flex-col items-center justify-center"
    >
      <div
        className="text-center p-4 w-full md:w-2/3"
      >
        <h3
          className="text-white text-lg md:text-2xl md:mb-10 font-bold"
          data-testid="h3"
        >
          YOUR RESERVED CARS
        </h3>
        <p
          className="text-white mt-5 md:text-xl"
          data-testid="p"
        >
          You can manage your cars here.
        </p>
      </div>
      <div
        className="p-1 border overflow-hidden border-white rounded-md mx-4 md:mt-10"
      >
        <div>
          <div
            className="shadow-md bg-white"
          >
            <table
              className="w-auto flex flex-col"
              data-testid="table"
            >
              <thead>
                <tr
                  className="flex space-x-5 w-full h-auto bg-gray-800 p-2 text-gray-200"
                >
                  <th
                    className="text-sm w-1/4"
                  >
                    Name
                  </th>
                  <th
                    className="text-sm w-1/4"
                  >
                    Start Date
                  </th>
                  <th
                    className="text-sm w-1/4"
                  >
                    End Date
                  </th>
                  <th
                    className="text-sm w-1/4"
                  >
                    Duration
                  </th>
                  <th
                    className="text-sm w-1/4"
                  >
                    Actions
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  className="flex item-center space-x-5 w-full p-2 text-center"
                >
                  <td
                    className="w-full text-center"
                    colSpan={3}
                  >
                    No reservation available
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
